app/src/main/java/com/github/wnder/TakePictureFragment.java

```diff
@@ -104,35 +104,52 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         user = GlobalUser.getUser();
         userName = user.getName();
 
+        if(checkIfFragmentIsAvailable(view)){
+            initPermAndCamera();
+        }
+    }
+
+    /**
+     * Checks if the take picture fragment should be available to user
+     * @param view the fragment's view
+     * @return true if available, false otherwise
+     */
+    private boolean checkIfFragmentIsAvailable(View view) {
```

Comment:
It should probably just be called "checkIfUserCanUploadPicture" or "IsAllowedToUpload" or something like that, because this make it feel like it check if the fragement could be unavailable because of missing ressource or something similar.

Reply:
changed to "checkIfUserCanUploadPicture"